aiida/tools/visualization/graph.py

```diff
@@ -471,9 +467,58 @@ def add_edge(self, in_node, out_node, link_pair=None, style=None, overwrite=Fals
         _add_graphviz_edge(self._graph, in_node, out_node, style)
 
     @staticmethod
-    def _convert_link_types(link_types):
-        """ convert link types, which may be strings, to a member of LinkType
+    def _link_types_to_traversal_rules(link_types, direction):
+        """convert link types list to traversal rules
+
+        :type link_types: str or :py:class:`aiida.common.links.LinkType`
+        :param link_types: link (or list of links) to be traversed by the resulting rules.
+        :param str direction: 'forward' (descendants) or 'backward' (ancestors).
         """
+
+        if direction not in ['forward', 'backward']:
+            raise ValueError('Direction must be `forward` or `backward`')
+
+        # For visualization allow toggling of all rules, with the default settings yielding a minimal graph
+        default_traversal_rules = GraphTraversalRules.DEFAULT.value
+
+        if not link_types:
+            return {name: direction in name for name in default_traversal_rules}
+
+        if isinstance(link_types, str):
+            link_types = [link_types]
+
+        if isinstance(link_types, LinkType):
+            link_types = [link_types.type]
+
+        if not isinstance(link_types, list):
+            raise ValueError(
+                'link_types must be a LinkType, a string, or a list of those.\ninstead, it is: {}'.format(link_types)
+            )
+
+        for indx, link_type in enumerate(link_types):
+            if isinstance(link_type, LinkType):
+                link_types[indx] = link_type.type
+            if not isinstance(link_type, str):
+                raise ValueError(
+                    'elements of link_types must be a LinkType or a string.\ninstead, one of them is: {}'.
+                    format(link_types)
+                )
+
+        names = ['_'.join([link_type, direction]) for link_type in link_types]
+
+        traversal_rules = {}
+        for name in names:
+            if name not in default_traversal_rules:
+                raise ValueError('{} is not a valid traversal rule'.format(name))
+
+        for name in default_traversal_rules:
+            traversal_rules[name] = name in names
```

Comment:
These can probably be joined in a single loop right? That being said, shouldn't the validation of traversal rules be done in `traverse_graph` call? Why do it again here

Reply:
Well, not exactly, `traverse_graph` no longer checks rules because it just receives links directly, now `validate_traversal_rules` is the one that checks the rules. Anyways, I was trying to not touch this method since anyways I wanted to eventually remove it (because it was basically translating link types to traversal rules, to be re-translated to link types by `validate_traversal_rules`), so since it would anyways need modifications I decided to remove it now.